pkg/codegen/docs/gen.go

```diff
@@ -479,15 +488,40 @@ func (mod *modContext) getProperties(properties []*schema.Property, lang string,
 		}
 
 		characteristics := propertyCharacteristics{
-			input:    isInput,
+			input:    input,
 			optional: !prop.IsRequired,
 		}
+
+		propLangName := prop.Name
+		switch lang {
+		case "python":
+			pyName := python.PyName(prop.Name)
+			// The default casing for a Python property name is snake_case unless
+			// it is a property of a nested object, in which case, we should check the property
+			// case maps.
+			propLangName = pyName
+
+			if nested {
```

Comment:
@justinvp I changed the logic here to only check the property case maps for Python when the properties that are passed in are nested properties. This makes it so that top-level properties use snake_case, by default, except for nested properties. This makes it so that for Functions, for which we do not build a separate property case map (neither does the Python code gen), the properties are snake_case'd except for nested properties, which will honor the case maps.